product_supplier_code_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-17 Eficent Business and IT Consulting Services, S.L.
+#           (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    @api.multi
+    def _compute_product_supplier_code(self):
+        product_supplierinfo_obj = self.env['product.supplierinfo']
+        for line in self:
+            partner = line.order_id.partner_id
+            product = line.product_id
+            if product and partner:
+                supplier_info = product_supplierinfo_obj.search([
```

Comment:
IMHO you shouldn't do that. On big purchases, your search requests is linearly proportional to your lines amount.

Please use a depends : 
```python
@api.depends('product_id.seller_ids.name', 'product_id.seller_ids.product_code')
```
ORM would do its job and load all you need in self.

Then iterate on self and get the right supplierinfo with a 'filtered':
```python
supplierinfo = line.seller_ids.filtered(lambda s, l=line: s.name == line.partner_id)
```